test/forms/buttons/button.test.js

```diff
@@ -1,136 +1,145 @@
+
+// test('Button is aria-disabled when form is invalid', () => {
+//   const wrapper = shallow(<Button invalid={true}> Hi</Button>)
+//   expect(wrapper.props()['aria-disabled']).toBe(true)
+// })
 
 test('Button is aria-disabled when form is invalid', () => {
-  const wrapper = shallow(<Button invalid={true}> Hi</Button>)
-  expect(wrapper.props()['aria-disabled']).toBe(true)
+  render(<Button invalid={true}>Hi</Button>)
+  expect(screen.getByRole('button')).toBeInTheDocument()
+  expect(screen.getByRole('button')).toHaveAttribute('aria-disabled')
 })
 
 test('Button is aria-disabled when form is pristine', () => {
-  const wrapper = shallow(<Button pristine={true}> Hi</Button>)
-  expect(wrapper.props()['aria-disabled']).toBe(true)
+  render(<Button pristine={true}> Hi</Button>)
+
+  expect(screen.getByRole('button')).toBeInTheDocument()
+  expect(screen.getByRole('button')).toHaveAttribute('aria-disabled')
 })
 
 test('Button onClick is run when the form is not submitting, pristine, or invalid', () => {
   const onClick = jest.fn()
+
   const formProps = {
     invalid: false,
     pristine: false,
     submitting: false,
   }
-  const wrapper = shallow(
+  render(
     <Button onClick={onClick} {...formProps}>
       {' '}
       Hi
     </Button>
   )
-  wrapper.find('button').simulate('click')
+  const button = screen.getByRole('button')
+
+  fireEvent.click(button)
 
   expect(onClick).toHaveBeenCalled()
 })
 
 test('Button onClick is not run when form is invalid', () => {
   const onClick = jest.fn()
-  const wrapper = mount(
+  render(
     <Button onClick={onClick} invalid={true}>
       {' '}
       Hi
     </Button>
   )
-  wrapper.find('button').simulate('click')
+  const button = screen.getByRole('button')
+
+  fireEvent.click(button)
 
   expect(onClick).not.toHaveBeenCalled()
 })
 
 test('Button onClick is not run when form is pristine', () => {
   const onClick = jest.fn()
-  const wrapper = mount(
+  render(
     <Button onClick={onClick} pristine={true}>
       {' '}
       Hi
     </Button>
   )
-  wrapper.find('button').simulate('click')
+
+  const button = screen.getByRole('button')
+
+  fireEvent.click(button)
 
   expect(onClick).not.toHaveBeenCalled()
 })
 
 test('Button onClick is not run when form is submitting', () => {
   const onClick = jest.fn()
-  const wrapper = mount(
+  render(
     <Button onClick={onClick} submitting={true}>
       {' '}
       Hi
     </Button>
   )
-  wrapper.find('button').simulate('click')
+
+  const button = screen.getByRole('button')
+
+  fireEvent.click(button)
 
   expect(onClick).not.toHaveBeenCalled()
 })
 
 test('Button adds variant string to class', () => {
-  const wrapper = shallow(<Button variant="custom"> Hi</Button>)
-  expect(wrapper.hasClass('button-custom')).toBe(true)
+  render(<Button variant="custom"> Hi</Button>)
+  expect(screen.getByRole('button')).toBeInTheDocument()
+  expect(screen.getByRole('button').getAttribute("class")).toContain("custom")
```

Comment:
In this case, it might be safer to check for the full class name (`button-custom`). Devs might be relying upon the `.button-custom` class in their CSS.

```suggestion
  expect(screen.getByRole('button')).toHaveClass('button-custom')
```